nats.go

```diff
@@ -1083,14 +1086,11 @@ func (nc *Conn) processMsg(data []byte) {
 
 	sub.mu.Lock()
 
-	// This is a catch all for more than max messages delivered.
-	if sub.max > 0 && sub.msgs > sub.max {
-		sub.mu.Unlock()
-		nc.removeSub(sub)
-		nc.mu.Unlock()
-		return
-	}
-
```

Comment:
Are you saying that there is a race and this is why you remove or you get the race after removing? Could you please include the race report?

The point of this check was to make sure that the connection does not "deliver" (that is pass it to the deliverMsgs routing or NextMsg) more messages than the set max number. 
Max can be changed, but that would be done under sub.mu.lock(), so we are fine.